pkg/kubelet/prober/prober_test.go

```diff
@@ -333,3 +333,138 @@ func TestNewExecInContainer(t *testing.T) {
+		{"runner set", prober.runner, runner},
+		{"recorder set", prober.recorder, recorder},
+		{"exec probe initialized", prober.exec != nil, true},
+		{"http probe initialized", prober.http != nil, true},
+		{"tcp probe initialized", prober.tcp != nil, true},
+		{"grpc probe initialized", prober.grpc != nil, true},
+	}
+
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			if tt.got != tt.expect {
+				t.Errorf("Expected %v, but got %v", tt.expect, tt.got)
+			}
+		})
+	}
+}
+
+func TestExecInContainer_Start(t *testing.T) {
+	ctx := ktesting.Init(t)
+	runner := &containertest.FakeContainerCommandRunner{
+		Stdout: "executed",
+	}
+	prober := &prober{
+		runner: runner,
+	}
+
+	container := v1.Container{}
+	containerID := kubecontainer.ContainerID{Type: "docker", ID: "containerID"}
+	cmd := []string{"ls", "-l"}
+
+	exec := prober.newExecInContainer(ctx, container, containerID, cmd, 0)
+	err := exec.Start()
+	require.NoError(t, err, "Expected no error, got %v", err)
+
+	err = exec.Wait()
+	require.NoError(t, err, "Expected no error on Wait, got %v", err)
+
+}
+
+func TestRecordContainerEventUnknownStatus(t *testing.T) {
+
+	if err := v1.AddToScheme(legacyscheme.Scheme); err != nil {
+		t.Errorf("failed to add v1 to scheme: %v", err)
+	}
+
+	pod := &v1.Pod{
+		ObjectMeta: metav1.ObjectMeta{
+			UID: "test-probe-pod",
+		},
+		Spec: v1.PodSpec{
+			Containers: []v1.Container{
+				{
+					Name: "test-probe-container",
+				},
+			},
+		},
+	}
+
+	container := pod.Spec.Containers[0]
+	recorder := record.NewFakeRecorder(10)
+
+	pb := &prober{
+		recorder: recorder,
+	}
+
+	output := "probe output"
+
+	testCases := []struct {
+		probeType probeType
+		result    probe.Result
+		expected  []string
+	}{
+		{
+			probeType: readiness,
+			result:    probe.Unknown,
+			expected: []string{
+				"Warning ContainerProbeWarning Readiness probe warning: probe output",
+				"Warning ContainerProbeWarning Unknown Readiness probe status: unknown",
+			},
+		},
+		{
+			probeType: liveness,
+			result:    probe.Unknown,
+			expected: []string{
+				"Warning ContainerProbeWarning Liveness probe warning: probe output",
+				"Warning ContainerProbeWarning Unknown Liveness probe status: unknown",
+			},
+		},
+		{
+			probeType: startup,
+			result:    probe.Unknown,
+			expected: []string{
+				"Warning ContainerProbeWarning Startup probe warning: probe output",
+				"Warning ContainerProbeWarning Unknown Startup probe status: unknown",
+			},
+		},
+	}
+
+	for _, tc := range testCases {
+		pb.recordContainerEvent(pod, &container, v1.EventTypeWarning, "ContainerProbeWarning", "%s probe warning: %s", tc.probeType, output)
+		pb.recordContainerEvent(pod, &container, v1.EventTypeWarning, "ContainerProbeWarning", "Unknown %s probe status: %s", tc.probeType, tc.result)
+
+		// Create a slice to hold event messages
+		var events []string
+
+		// Receive events from the channel
+		for i := 0; i < 2; i++ {
+			events = append(events, <-recorder.Events)
```

Comment:
Would this block the test if second even is not received for some reason?

Reply:
Indeed - tried to create a buffered channel to avoid blocking. Is that a good approach ?

Reply:
I don't know. Have you tested if it helps? How similar tests work with `FakeRecorder`?

Reply:
@jralmaraz This one is still not addressed. Have you looked at other tests that use FakeRecorder? Do they use the same approach?

Reply:
Hi @bart0sh - sorry for leaving this one behind. I have changed the implementation to created the `FakeRecorder` with the buffer size based on the length of the expected events in the test cases. Is that an ok approach ? 

I checked the [cronjob_controllerv2_test](https://github.com/kubernetes/kubernetes/blob/master/pkg/controller/cronjob/cronjob_controllerv2_test.go#L1321) , [daemon_controller_test](https://github.com/kubernetes/kubernetes/blob/master/pkg/controller/daemon/daemon_controller_test.go#L331) and a few others, but still trying to learn how to best use it.

Reply:
> Is that an ok approach ?

I'm not sure it's ok. I'd at least check if amount of events is expected before reading them.

BTW, you can comment out one of of the `pb.recordContainerEvent` calls to see that making a bigger buffer doesn't make reading from an empty channel successful.

Reply:
Currently commenting it was timing out, so I added a check to fail before, but still working on it and reviewing before pushing again.